src/box/lua/schema.lua

```diff
@@ -411,6 +414,12 @@ box.begin = function(options)
         if txn_isolation ~= nil then
             txn_isolation = normalize_txn_isolation_level(txn_isolation)
         end
+        linearizable = options.linearizable
+        if linearizable ~= nil and
+           type(linearizable) ~= 'boolean' then
+            box.error(box.error.ILLEGAL_PARAMS,
+                      "linearizable must be a boolean")
+        end
```

Comment:
Good idea, haven't thought of that.
Done in a new commit:
<details><summary>diff</summary>

```diff
commit e9ca65ca3a7c71c786ca9a8fd11798037ea52161
Author: Serge Petrenko <sergepetrenko@tarantool.org>
Date:   Wed Sep 14 13:15:24 2022 +0300

    box: refactor box.begin() options validation
    
    Reuse check_param_table() funciton in box begin instead of some custom
    parameter checking code.
    As one of the side effects, unknown options are now declined by
    `box.begin{}`, which becomes more and more important with new options
    introduction.
    
    In-scope-of #6707
    
    NO_DOC=refactoring
    NO_CHANGELOG=refactoring

diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index 51a44d99e..864dd4a81 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -397,21 +397,22 @@ end
 
 box.internal.normalize_txn_isolation_level = normalize_txn_isolation_level
 
-box.begin = function(options)
-    local timeout
-    local txn_isolation
-    if options then
-        check_param(options, 'options', 'table')
-        timeout = options.timeout
-        if timeout and (type(timeout) ~= "number" or timeout <= 0) then
+local begin_options = {
+    timeout = function(timeout)
+        if type(timeout) ~= "number" or timeout <= 0 then
             box.error(box.error.ILLEGAL_PARAMS,
                       "timeout must be a number greater than 0")
         end
-        txn_isolation = options.txn_isolation
-        if txn_isolation ~= nil then
-            txn_isolation = normalize_txn_isolation_level(txn_isolation)
-        end
-    end
+        return true
+    end,
+    txn_isolation = normalize_txn_isolation_level,
+}
+
+box.begin = function(options)
+    check_param_table(options, begin_options)
+    local timeout = options and options.timeout
+    local txn_isolation = options and options.txn_isolation and
+                          normalize_txn_isolation_level(options.txn_isolation)
     if builtin.box_txn_begin() == -1 then
         box.error()
     end
diff --git a/test/box-luatest/begin_options_validation_test.lua b/test/box-luatest/begin_options_validation_test.lua
new file mode 100644
index 000000000..b044dd610
--- /dev/null
+++ b/test/box-luatest/begin_options_validation_test.lua
@@ -0,0 +1,23 @@
+local server = require("test.luatest_helpers.server")
+local t = require("luatest")
+
+local g = t.group("begin-options-validation")
+
+g.before_all = function()
+    g.server = server:new{
+        alias = "default",
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_begin_options_validation = function()
+    g.server:exec(function()
+        require("luatest").assert_error_msg_content_equals(
+            "Illegal parameters, unexpected option 'foo'",
+            box.begin, {foo = "bar"})
+    end)
+end

```
</details>